src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java

```diff
@@ -240,6 +240,7 @@ public SkyValue compute(SkyKey skyKey, Environment env)
               return null;
             }
             if (repoState instanceof DigestWriter.RepoDirectoryState.UpToDate) {
+              candidate.touch();
```

Comment:
Move this further down to avoid touching multiple times on restarts.

Reply:
done.